### PR TITLE
SDK V2 docs

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
Adds SDK V2 docs to replace existing one and a migration guide
https://deploy-preview-11031--aptos-developer-docs.netlify.app/sdks/ts-sdk/index

Note:
our docs points to the ts sdk v1 code and examples all over. This PR changes to point to v2 where the docs just mentions ts sdk and when needed to not break the site.

Is there a page I can move V1 docs to? I would really love to get the V1 docs somewhere else (and link to it on v2 docs). Currently it is on the sidebar under the TypeScript SDK section, but the flow is annoying - i.e when reaching the last "[Typescript](https://deploy-preview-11031--aptos-developer-docs.netlify.app/sdks/ts-sdk/typescript-sdk-typescript)" section, the site offers you the first V1 docs page as the "next". 

Next PRs
- update e2e dapp tutorial to use sdk v2
- port over v1 examples to v2 and update docs accordingly
- end goal is for the docs to only refer to sdk v2

### CODE REVIEW

**Review by davidiw — COMMENTED** [review]
there seems to have been an auto-formatter run. Could you either make a separate PR / commit or revert those changes?

File: developer-docs-site/docs/sdks/ts-sdk/key-management.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+---
+title: "Key Management"
+slug: "typescript-sdk-key-management"
```
Comment (davidiw) [inline developer-docs-site/docs/sdks/ts-sdk/key-management.md:3]: please don't introduce slugs, let's just use the file system to designate routes.

**Comment by 0xmaayan** [comment]
> > there seems to have been an auto-formatter run. Could you either make a separate PR / commit or revert those changes?
> 
> hmm I ran `pnpm run fmt` - I was hoping that should format everything correctly..?

File: developer-docs-site/docs/sdks/ts-sdk/key-management.md (hunk 1 above)
Comment (0xmaayan) [reply developer-docs-site/docs/sdks/ts-sdk/key-management.md]: removed

File: developer-docs-site/docs/sdks/ts-sdk/index.md [hunk 2]
```diff
@@ -4,46 +4,117 @@ slug: "index"
...
+import ThemedImage from '@theme/ThemedImage';
+
 # Aptos TypeScript SDK
 
-Aptos provides a fully supported TypeScript SDK with the source code in the [Aptos-core GitHub](https://github.com/aptos-labs/aptos-core/tree/main/ecosystem/typescript/sdk) repository. Much of the functionality of the TypeScript SDK can be found in the [Rust](../rust-sdk.md) and [Python](../python-sdk.md) SDKs. Nevertheless, Aptos strongly encourages you to use the TypeScript SDK for app development whenever possible.
+:::note
+This documentation is for TypeScript SDK > ` v2.0.0`. You can find the documentation for SDK `v1.x.x` [here](./v1/index.md)
+
+Looking to migrate to `TypeScript SDK v2`? check out the [migration guide](./migration-guide.md)
+:::
```
Comment (davidiw) [inline developer-docs-site/docs/sdks/ts-sdk/index.md]: Hmm... this should only be temporary while and we should ideally take it down in 2 weeks or so imo.

File: developer-docs-site/docs/sdks/ts-sdk/key-management.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
+---
+title: "Key Management"
+---
+
+The SDK provides an `Account` class for creating and managing [account](../../concepts/accounts.md) on Aptos network.
```
Comment (davidiw) [inline developer-docs-site/docs/sdks/ts-sdk/key-management.md]:
```suggestion
The SDK provides an `Account` class for creating and managing [accounts](../../concepts/accounts.md) on Aptos network.
```

File: developer-docs-site/docs/sdks/ts-sdk-v2/index.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
+---
+title: "TypeScript Index"
+slug: "index"
+hidden: false
```
Comment (davidiw) [inline developer-docs-site/docs/sdks/ts-sdk-v2/index.md:4]: remove

File: developer-docs-site/docs/sdks/ts-sdk-v2/key-management.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
+---
+title: "Key Management"
+---
+
+The SDK provides an `Account` class for creating and managing [account](../../concepts/accounts.md) on Aptos network.
```
Comment (davidiw) [inline developer-docs-site/docs/sdks/ts-sdk-v2/key-management.md]:
```suggestion
The SDK provides an `Account` class for creating and managing [accounts](../../concepts/accounts.md) on Aptos network.
```

File: developer-docs-site/docs/sdks/ts-sdk-v2/move-types.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+When developing on Aptos, and specifically working with the SDK, developers often need to handle Move types serialization and deserialization. Whether is to construct a transaction payload, build a raw transaction or read BCS data.
+
+The SDK provides a convenient Move sub-classes to easily interact with move types to perform serialization or deserialization operations.
+Each class has a `serialize`, `serializeForEntryFunction` and `serializeForScriptFunction` methods and a `deserialize` static class.
+
+In addition, for complex types like `Vector` the SDK supports nested serialization and deserialization.
+
+### Move primitive types
+
+Classes to handle Move primitive types:
```
Comment (davidiw) [inline developer-docs-site/docs/sdks/ts-sdk-v2/move-types.md:16]: what about address?

**Review by davidiw — COMMENTED** [review]
really quick review, will follow up in a bit

I don't think it needs much more than the recommendations, so we can launch after the said fixes

File: developer-docs-site/docs/sdks/ts-sdk/index.md (hunk 2 above)
Comment (0xmaayan) [reply developer-docs-site/docs/sdks/ts-sdk/index.md]: you mean take this `note` down....?

File: developer-docs-site/docs/sdks/ts-sdk-v2/move-types.md (hunk 6 above)
Comment (0xmaayan) [reply developer-docs-site/docs/sdks/ts-sdk-v2/move-types.md:16]: we use `AccountAddress` class for that

Comment (davidiw) [reply developer-docs-site/docs/sdks/ts-sdk-v2/move-types.md:16]: address is actually a move primitive, so its exclusion here is awkward (to me)

File: developer-docs-site/docs/sdks/ts-sdk/index.md (hunk 2 above)
Comment (davidiw) [reply developer-docs-site/docs/sdks/ts-sdk/index.md]: yeah, just a mental comment, so we do remove it...

File: developer-docs-site/docs/sdks/ts-sdk-v2/move-types.md (hunk 6 above)
Comment (0xmaayan) [reply developer-docs-site/docs/sdks/ts-sdk-v2/move-types.md:16]: gotcha. I'll simply add `AccountAddress` to this list and leave a todo comment to use a address move primitive class to extend from

File: developer-docs-site/docs/sdks/ts-sdk/index.md [hunk 7]
```diff
@@ -6,6 +6,10 @@ hidden: false
 
 # Aptos TypeScript SDK
 
+:::note
+This documentation is for TypeScript SDK `v1.x.x`. For a more robust and better SDK support, we recommend upgrading to `v2.0.0`. Take a look at the [TypeScript SDK v2 documentation](../ts-sdk-v2/index.md) and the [migration guide](../ts-sdk-v2/migration-guide.md)
```
Comment (gregnazario) [inline developer-docs-site/docs/sdks/ts-sdk/index.md:10]: v2.x.x

File: developer-docs-site/docs/sdks/ts-sdk-v2/fetch-data-from-chain.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+1. Query for the ledger info
+
+```ts
+const ledgerInfo = await aptos.getLedgerInfo();
+
+const ledgerVersion = ledgerInfo.ledger_version;
+```
+
+2. If we just commited a transaction with the SDK, we can use `waitForTransaction` method, that would return us a `CommittedTransactionResponse` that holds the latest ledger version
```
Comment (gregnazario) [inline developer-docs-site/docs/sdks/ts-sdk-v2/fetch-data-from-chain.md:85]:
```suggestion
2. If we just committed a transaction with the SDK, we can use `waitForTransaction` method, that would return us a `CommittedTransactionResponse` that holds the latest ledger version
```

**Review by gregnazario — APPROVED** [review]

File: developer-docs-site/docs/sdks/ts-sdk-v2/fetch-data-from-chain.md (hunk 8 above)
Comment (0xmaayan) [reply developer-docs-site/docs/sdks/ts-sdk-v2/fetch-data-from-chain.md:85]: oh wow, you are better then `pnpm run spellcheck`

**Review by davidiw — APPROVED** [review]
:crossed_fingers:

**Pull request merged**